security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt

```diff
@@ -208,38 +195,33 @@ class AesEncryptionManager(
                 }
             }
 
-    override fun observeFingerprintForUnlock(): Observable<FingerprintUnlockResult> =
-        Observable
-            .fromCallable {
-                val cryptedData = preferencesManager.prefs.getString(PREF_KEY_FINGERPRINT_ENCRYPTED_APP_KEY, null) ?: throw FingerprintUnlockError()
-                CryptoData.fromString(cryptedData)
-            }
-            .flatMap { cryptedData -> fingerprintHelper.authenticate(cryptedData.iv).map { cryptedData to it } }
-            .subscribeOn(Schedulers.io())
-            .map { (cryptedData, authResult) ->
-                when (authResult) {
-                    is AuthenticationResultSuccess -> {
-                        synchronized(keyLock) {
-                            key = authResult.cipher.doFinal(cryptedData.data)
-                        }
-                        if (key != null) FingerprintUnlockSuccessful() else throw FingerprintUnlockError()
-                    }
-                    is AuthenticationFailed -> FingerprintUnlockFailed()
-                    is AuthenticationHelp -> FingerprintUnlockHelp(authResult.helpString)
+    override suspend fun unlockWithFingerprint(): FingerprintUnlockResult {
+        val cryptedData = CryptoData.fromString(
+            preferencesManager.prefs.getString(PREF_KEY_FINGERPRINT_ENCRYPTED_APP_KEY, null) ?: throw FingerprintUnlockError()
+        )
+        val authResult = fingerprintHelper.authenticate(cryptedData.iv)
```

Comment:
In case you didn't know, you can put this whole line within the parenthesis of the `when()` :D

Reply:
I just copy pasted code around without looking at the style

Reply:
I want to do this.